### PR TITLE
ensure video resets state correctly when a new src is loaded

### LINKED ISSUE #2651: Changing from video input example while video is playing breaks
### Describe the bug

In https://huggingface.co/spaces/sayakpaul/video-classification-ucf101-subset, I can easily switch between examples. If I select example 1, click play, and then switch to another example while it's running, it will break as I won't be able to play the video. Clicking pause/play does not seem to work

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Check out https://huggingface.co/spaces/sayakpaul/video-classification-ucf101-subset

### Screenshot

_No response_

### Logs

```shell
-
```

### System Info

```shell
Chrome, Mac OS, running in https://huggingface.co/spaces/sayakpaul/video-classification-ucf101-subset
```

### Severity

serious, but I can work around it

### LINKED ISSUE #3119: Video ui is jumpy when the value is changed
### Describe the bug

This video shows the issue:

https://user-images.githubusercontent.com/12937446/216451342-492e8d36-1f9d-4d23-a64c-abdaa8aa8a39.mov

I think the video is jumping from video size -> default size -> video size. It would be good if it at least didn't jump down to that inbetween size.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

https://user-images.githubusercontent.com/12937446/216451342-492e8d36-1f9d-4d23-a64c-abdaa8aa8a39.mov

### Screenshot

https://user-images.githubusercontent.com/12937446/216451342-492e8d36-1f9d-4d23-a64c-abdaa8aa8a39.mov

### Logs

```shell
no
```

### System Info

```shell
no
```

### Severity

annoying

### PR DESCRIPTION
**Pull request by pngwn**
~~Need to repro on spaces, struggling locally. I think it's a race condition.~~

Details in comments. Also fixed the annoying 'pause interrupts play' error in the console.

Closes #2651. Closes #3119.

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by pngwn** [comment]
Okay I've reproed this. Bunch of weird stuff happening. 

- The buttons aren't updating to match the actual play status
- The 'current progress' is an absurd number usually far longer than the actual video
- It isn't actually broken, the progress is just at the the end of the video and the play buttons aren't matching the state. Clicking back in the progress bar + pressing play/pause a few times get things back into the correct state.

Video when broken for posterity:

https://user-images.githubusercontent.com/12937446/216449266-8013bf81-47b9-4e6d-90f6-f87ecbc3b869.mov

**Comment by pngwn** [comment]
race condition as expected. fixed:


https://user-images.githubusercontent.com/12937446/216451342-492e8d36-1f9d-4d23-a64c-abdaa8aa8a39.mov

File: demo/video_component/run.py [hunk 1]
```diff
@@ -1,8 +1,30 @@
-import gradio as gr 
+import gradio as gr
+import os
 
-css = "footer {display: none !important;} .gradio-container {min-height: 0px !important;}"
+# css = (
```
Comment (freddyaboulton) [inline demo/video_component/run.py:4]: Can we delete this?

**Review by freddyaboulton — APPROVED** [review]
Confirmed the fix!

Awesome - thanks @pngwn !!

File: demo/video_component/run.py (hunk 1 above)
Comment (pngwn) [reply demo/video_component/run.py:4]: Oh yeah. Forgot about that. Nice catch.

**Comment by abidlabs** [comment]
It looks like this should fix https://github.com/gradio-app/gradio/issues/2651 as well, right?

**Comment by pngwn** [comment]
@abidlabs Yeah, that issue is linked in the PR description. Not sure if there is anything else it will fix.

**Comment by abidlabs** [comment]
Oh sorry I meant #3119 or is that a different issue?

**Comment by pngwn** [comment]
Ah, no. I just created that issues as a result of this PR. It looks slightly annoying so I'd rather do it in a new PR.

**Comment by pngwn** [comment]
Just took a look at it and i think I've fixed it.

**Comment by pngwn** [comment]
Not perfect but much better now

https://user-images.githubusercontent.com/12937446/216580327-86688864-917e-4105-8d58-01ddac296645.mov

Fixed the jumpy resizing and reduced some of the flashing that was occurring due to different loading states.

**Comment by pngwn** [comment]
If folks could gibe this another quick test, I'll merge it in.

**Pull request merged**